### ISSUE TITLE
Customer/employee name should be a single field

### ISSUE THREAD

**Issue by Traumflug** [thirtybees/thirtybees#907]
Handling names of humans is a pretty tricky matter. Suggested W3C solution is to use just a single field. See

https://www.w3.org/International/questions/qa-personal-names.en
https://shinesolutions.com/2018/01/08/falsehoods-programmers-believe-about-names-with-examples/

For thirty bees, a customer name is just some string to print onto invoices and shipping labels, so no need to enforce this to have multiple fields.

That said, having multiple name fields is pretty deeply engraved into the name handling code, so a proper solution needs some intelligent backwards compatibility strategy.

**Comment by Dh42** [comment]
This will destroy the ability for merchants to do name verification on any kind of credit card charges. AVS systems use an exact match on credit card issue names to figure out if they are valid or not. We would need to do some fancy mechanics to try to extract the names and send them to the gateways.

**Comment by Traumflug** [comment]
Good point. Current code is probably better in this regard, but also disallows some classes of names. In some cultures, there is no such thing as a distinction between first and last name.
